R/utils.R

```diff
@@ -13,6 +13,42 @@ setup_project <- function() {
   source(path)
 }
 
-use_r_packages <- function(path = "deduplicate/load-and-attach-r-packages.R") {
+use_r_packages <- function(path = packages_path()) {
   source(path)
 }
+
+packages_path <- function() {
+  "deduplicate/load-and-attach-r-packages.R"
+}
+
+create_install_calls_for_dockerfile <- function(path = packages_path()) {
+  pkg <- sub("library\\((.*)\\)", "\\1", readLines(path, encoding = "UTF-8"))
+  glue('    && Rscript -e "install.packages(\'{pkg}\')" \\')
+}
+
+update_dockerfile_packages <- function(path = "docker/Dockerfile") {
+  dkr <- readLines(path, encoding = "UTF-8")
+
+  raw <- readLines(packages_path(), encoding = "UTF-8")
+  pkg <- sub("library\\((.*)\\)", "\\1", raw)
+  pkg <- glue::glue('    && Rscript -e "install.packages(\'{pkg}\')" \\')
+  # pkg <- sprintf('    && Rscript -e "install.packages(\'%s\')" \\', pkg)
+
+  out <- c(
+    dkr[1:marker_start(dkr)],
+    pkg,
+    dkr[marker_end(dkr):length(dkr)]
+  )
+  writeLines(out, path)
+
+  invisible(path)
+}
```

Comment:
This function helps update the calls to `Rscript -e "install.package(...)" in the Dockerfile. It overwrites the existing
Dockerfile with packages detected in "deduplicate/load-and-attach-r-packages.R".